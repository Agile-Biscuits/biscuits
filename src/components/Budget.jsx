import styled from '@emotion/styled';

const getFillHeight = ({ amount, value }) => {
  const percentage = (value / amount);

  return percentage > 1 ? 198 : percentage * 200 - 2;
};
const getFillColor = ({
  amount, value,
}) => {
  const ratio = value / amount;

  if (ratio > 1) return 'cornflowerblue';

  return value / amount > 0.2
    ? 'lightseagreen'
    : '#FF6347';
};

const Container = styled.div`
  position: relative;
  display: flex;
  width: 92%;
  height: 100px;
  background: whitesmoke;
  border: 1px solid whitesmoke;
  border-radius: 8px;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25) inset;
  overflow: hidden;

  &::before {
    position: absolute;
    top: 16px;
    left: 12px;
    font-size: 20px;
    font-weight: 500;
    color: #182B2B;
    text-shadow: -1px -1px 0 rgba(255,255,255,0.2),
      1px -1px 0 rgba(255,255,255,0.2),
      -1px 1px 0 rgba(255,255,255,0.2),
      1px 1px 0 rgba(255,255,255,0.2);
    content: '${({ name }) => name}';
  }

  &::after {
    position: absolute;
    bottom: 12px;
    right: 12px;
    font-size: 36px;
    font-weight: 800;
    color: #182B2B;
    text-shadow: -1px -1px 0 rgba(255,255,255,0.2),
      1px -1px 0 rgba(255,255,255,0.2),
      -1px 1px 0 rgba(255,255,255,0.2),
      1px 1px 0 rgba(255,255,255,0.2);
    content: '${({ value }) => value}';
  }
`;
const Fill = styled.div`
  align-self: flex-end;
  width: 100%;
  height: ${getFillHeight}px;
  background: ${getFillColor};
  border: 1px solid rgba(255,255,255,0.2);
  border-radius: 7px;
  box-shadow: 0 -1px 4px rgba(0, 0, 0, 0.25);
`;

export default function Budget({ name, amount, value }) {
  return (
    <Container name={name} value={value}>
      <Fill amount={amount} value={value} />
    </Container>
  );
}
