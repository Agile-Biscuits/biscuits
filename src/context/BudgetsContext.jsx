import { createContext, useState, useEffect } from 'react';
import budgets from '../../fixtures/budgets';

export const BudgetsContext = createContext();

export const BudgetsProvider = ({ children }) => {
  const [budgetsData, setBudgetsData] = useState([]);

  useEffect(() => {
    setBudgetsData(budgets); // set your budgets data when the component mounts
  }, []);

  return (
    <BudgetsContext.Provider value={{ budgets: budgetsData, setBudgets: setBudgetsData }}>
      {children}
    </BudgetsContext.Provider>
  );
};
