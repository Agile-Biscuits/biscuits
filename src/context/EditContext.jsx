import { createContext, useState } from 'react';

export const EditContext = createContext({
  isEditing: false,
  setIsEditing: () => {}
});

export const EditProvider = ({ children }) => {
  const [isEditing, setIsEditing] = useState(false);

  return (
    <EditContext.Provider value={{ isEditing, setIsEditing }}>
      {children}
    </EditContext.Provider>
  );
};
