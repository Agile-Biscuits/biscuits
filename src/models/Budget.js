export default class Budget {
  constructor({ name, amount, value, id }) {
    this.name = name;
    this.amount = amount;
    this.value = value;
    this.id = id;
  }
}
