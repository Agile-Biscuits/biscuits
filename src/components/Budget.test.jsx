import { render, cleanup } from '@testing-library/react';
import Budget from './Budget';
import { EditProvider } from '../context/EditContext';

afterEach(cleanup);

describe('Budget', () => {
  it('should render the budget', () => {
    expect(
      render(
        <EditProvider>
          <Budget name="Test" amount={100} value={50}/>
        </EditProvider>
      )
    ).toBeTruthy()
  });
  it('should render the budget name', () => {
    const { getByText } = render(
      <EditProvider>
        <Budget name="Test" amount={100} value={50}/>
      </EditProvider>
    );
    expect(getByText('Test')).toBeTruthy();
  });
  it('should render the budget amount', () => {
    const { getByText } = render(
      <EditProvider>
        <Budget name="Test" amount={100} value={50}/>
      </EditProvider>
    );
    expect(getByText('Â£50.00')).toBeTruthy();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(
      <EditProvider>
        <Budget name="Test" amount={100} value={50}/>
      </EditProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
