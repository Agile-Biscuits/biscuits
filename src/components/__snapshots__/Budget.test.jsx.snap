// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-8w9okj-Container e1dz7f3g7"
    data-testid="budget"
  >
    <div
      class="css-1qtj0e0-Fill e1dz7f3g0"
      value="50"
    />
    <div
      class="css-15xwpwu-InfoContainer e1dz7f3g6"
    >
      <div
        class="css-njdknn-NameContainer e1dz7f3g2"
      >
        <div
          class="css-kh8k75-Name e1dz7f3g5"
        >
          TEST
        </div>
      </div>
      <div
        class="css-7nkki1-Value e1dz7f3g1"
      >
        Â£50.00
      </div>
    </div>
  </div>
</DocumentFragment>
`;
